src/include/storage/layout.h

```diff
@@ -0,0 +1,144 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// layout.h
+//
+// Identification: src/include/storage/layout.h
+//
+// Copyright (c) 2015-2018, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <map>
+
+#include "common/internal_types.h"
+#include "common/printable.h"
+
+//@{
+/** @brief  Pre-defined OIDs for ROW and COLUMN store.  */
+#define ROW_STORE_OID 0
+#define COLUMN_STORE_OID 1
+//@}
+
+namespace peloton {
+
+namespace catalog {
+class Schema;
+} // namespace catalog
+
+namespace storage {
+
+/** @brief used to store the mapping between a tile and its columns
+ * <tile index> to vector{<original column index, tile column offset>}
+ */
+typedef std::map<oid_t, std::vector<std::pair<oid_t, oid_t>>> tile_map_type;
+
+/**
+ * @brief   Class to store the physical layout of a TileGroup.
+ */
+class Layout : public Printable {
+ public:
+  /**
+   * @brief   Constructor for predefined layout types.
+   * @param   num_columns Number of columns in the layouts.
+   * @param   layout_type Has to be LayoutType::ROW or LayoutType::COLUMN.
+   */
+  Layout(const oid_t num_columns, LayoutType layout_type = LayoutType::ROW);
+
+  /**
+   * @brief   Constructor for arbitrary column_maps.
+   *          If the column_map is of type LayoutType::HYBRID,
+   *          the layout_oid_ is set to INVALID_OID. For all other
+   *          layouts, the pre-defined OIDs are used. Should be used
+   *          only for testing & TempTables, the layout isn't persistent.
+   * @param   column_map Column map of the layout to be constructed
+   *
+   */
+  Layout(const column_map_type &column_map);
+
+  /**
+   * @brief   Constructor for arbitrary column_maps.
+   * @param   column_map Column map of the layout to be constructed.
+   * @param   layout_oid Per-table unique OID. Generted by DataTable.
+   */
+  Layout(const column_map_type &column_map, oid_t layout_oid);
+
+  /** @brief  Check whether this layout is a row store. */
+  bool IsRowStore() const { return (layout_type_ == LayoutType::ROW); }
+
+  /** @brief  Check whether this layout is a column store. */
+  bool IsColumnStore() const { return (layout_type_ == LayoutType::COLUMN); }
+
+  /** @brief  Return the layout_oid_ of this object. */
+  oid_t GetOid() const { return layout_oid_; }
+
+  /** @brief  Sets the tile id and column id w.r.t. of the tile corresponding
+   *          to the specified tile group column id.
+   */
+  void LocateTileAndColumn(oid_t column_offset, oid_t &tile_offset,
```

Comment:
Misnamed? More correctly something more like SetTileAndColumn?

Reply:
I am sorry! This function was previously a part of `TileGroup`, I copied the comments as is. 

I have changed it now. Essentially, we are using references here because we need to return both the `tile_id` and `tile_column_id`.